WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -200,12 +200,25 @@ public TransactionFactory(Network network, KeyManager keyManager, ICoinsView coi
 		{
 			totalOutgoingAmountNoFee = realToSend.Where(x => !changeHdPubKey.ContainsScript(x.destination.ScriptPubKey)).Sum(x => x.amount);
 		}
-		decimal totalOutgoingAmountNoFeeDecimal = totalOutgoingAmountNoFee.ToDecimal(MoneyUnit.BTC);
 
-		// Cannot divide by zero, so use the closest number we have to zero.
-		decimal totalOutgoingAmountNoFeeDecimalDivisor = totalOutgoingAmountNoFeeDecimal == 0 ? decimal.MinValue : totalOutgoingAmountNoFeeDecimal;
-		decimal feePercentage = 100 * fee.ToDecimal(MoneyUnit.BTC) / totalOutgoingAmountNoFeeDecimalDivisor;
+		decimal totalOutgoingAmountNoFeeDecimal = totalOutgoingAmountNoFee.ToDecimal(MoneyUnit.BTC);
+		decimal feeDecimal = fee.ToDecimal(MoneyUnit.BTC);
 
+		decimal feePercentage;
+		if (payments.ChangeStrategy == ChangeStrategy.AllRemainingCustom)
+		{
+			// In this scenario since the amount changes as the fee changes, we need to compare against the total sum / 2,
+			// as with this, we will make sure the fee cannot be higher than the amount.
+			decimal inputSumDecimal = spentCoins.Sum(x => x.Amount.ToDecimal(MoneyUnit.BTC));
+			feePercentage = 100 * (feeDecimal / (inputSumDecimal / 2));
```

Comment:
I don't understand the change. Why divided by 2?

Reply:
Currently, on master, we don't allow the build of a transaction where the fee is higher than the amount to be sent.
To have the same functionality in the case of `ChangeStrategy.AllRemainingCustom` we have to compare against the `input sum / 2` because that is the maximum fee that we allow.

Examples:
```
Input: 1 BTC
Fee: 0.2 BTC
Output: 0.8 BTC
feePercentage: 100 * (0.2 / (1 BTC / 2) = 40%

Input: 1 BTC
Fee: 0.5 BTC
Output: 0.5 BTC
feePercentage: 100 * (0.5 / (1 BTC / 2) = 100% // It pays the maximum that is allowed

Input: 1 BTC
Fee: 0.6 BTC
Output: 0.4 BTC
feePercentage: 100 * (0.6 / (1 BTC / 2) = 120% // Overpays as the fee is higher than the amount.
```

Note: In the case of self spend the total outgoing amount is 0, so the fee PC calculation allowed a transaction like this when the `ChangeStrategy` was `AllRemainingCustom`
![image](https://github.com/zkSNACKs/WalletWasabi/assets/16364053/09e80b78-3b5d-41da-9ff6-184aedfc18c1)